src/components/dashboard/Applications.tsx

```diff
@@ -3,14 +3,27 @@ import { useEffect, useMemo, useState } from "react";
 import { useStorage } from "@plasmohq/storage/hook";
 import { ExtensionStorage } from "~utils/storage";
 import { SettingsList } from "./list/BaseElement";
-import { useLocation, useRoute } from "wouter";
+import { useRoute } from "wouter";
 import Application from "~applications/application";
 import AppListItem from "./list/AppListItem";
 import browser from "webextension-polyfill";
 import SearchInput from "./SearchInput";
 import styled from "styled-components";
+import { useLocation } from "~wallets/router/router.utils";
+import type { CommonRouteProps } from "~wallets/router/router.types";
+
+export interface ApplicationsDashboardViewParams {
+  app?: string;
+}
+
+export type ApplicationsDashboardViewProps =
+  CommonRouteProps<ApplicationsDashboardViewParams>;
+
+export function ApplicationsDashboardView() {
```

Comment:
You will see most changes in the PR are like this file:

- Change default export to named export.
- Change exported component view to `<ComponentName>View`.
- Replace `wouter`'s `useLocation()` and/or `useHistory()` with the custom wrapper that extends its functionality and adds a `back()` function.
- When possible, get the route's params from the component's props (injected by the router) rather than using the `useRouter()` hook. In instances where that hasn't been changed, like here, it's because this view component is rendered manually (without using `<Routes>` / `<Route>`), which happens in some of the views of the Welcome and Dashboard apps still (only part of them have been refactored already).